src/Backend/SgateShopgatePlugin/Controllers/Frontend/shopgate.php

```diff
@@ -1006,6 +1281,654 @@ public function registrationAction()
+                        ->orderBy('items.id', 'DESC')
+                        ->setParameter('customerId', $userId)
+                        ->setParameter('shopId', $this->get('shop')->getId());
+
+                    $carts = $builder->getQuery()->getResult();
+
+                    if (empty($carts)) {
+                        $this->Request()->setPost('newlist', 'App');
+                    } else {
+                        $this->Request()->setPost('lists', array($carts[0]->getId()));
+                    }
+
+                    $this->Request()->setPost('ordernumber', $orderNumber);
+                    $this->container->get('swag_advanced_cart.cart_handler')->addToList($this->Request()->getPost());
+                } else {
+                    $productId = Shopware()->Modules()->Articles()->sGetArticleIdByOrderNumber($orderNumber);
+                    $productName = Shopware()->Modules()->Articles()->sGetArticleNameByOrderNumber($orderNumber);
+
+                    if (empty($productId)) {
+                        return false;
+                    }
+
+                    Shopware()->Modules()->Basket()->sAddNote($productId, $productName, $orderNumber);
+                }
+            } else {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * A helper function to check if the wish list from the advanced cart plugin is active
+     *
+     * @return bool
+     */
+    private function isWishList()
+    {
+        $pluginManager  = $this->container->get('shopware_plugininstaller.plugin_manager');
+        $plugin = $pluginManager->getPluginByName('SwagAdvancedCart');
+        $config = $this->container->get('shopware.plugin.cached_config_reader')->getByPluginName('SwagAdvancedCart');
+
+        if ($plugin->getInstalled() && $plugin->getActive() && $config['replaceNote']) {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * A helper function that prepares a single cart
+     *
+     * @param Cart $cart
+     *
+     * @return null|array
+     */
+    private function prepareCartItems($cart)
+    {
+        $items = array();
+
+        /** @var CartItem $cartItem */
+        foreach ($cart->getCartItems() as $cartItem) {
+            if (!$cartItem->getDetail() || !$cartItem->getDetail()->getActive()) {
+                continue;
+            }
+            $orderNumber = $cartItem->getProductOrderNumber();
+            $product = $this->prepareArticle($orderNumber, $cartItem);
+
+            if ($product) {
+                $items[] = $product;
+            }
+        }
+
+        return $items;
+    }
+
+    /**
+     * A helper function that returns a prepared product object ready to be displayed in the frontend
+     *
+     * @param int      $orderNumber
+     * @param CartItem $cartItem
+     *
+     * @return null|array
+     */
+    private function prepareArticle($orderNumber, $cartItem)
```

Comment:
I am assuming this function is related to the cart, should go to the Cart (create one if needed) class